addons/website/static/src/snippets/s_searchbar/000.js

```diff
@@ -1,127 +1,107 @@
-        'keydown .search-query, .dropdown-item': '_onKeydown',
-        'search .search-query': '_onSearch',
-    },
-    autocompleteMinWidth: 300,
-
-    /**
-     * @constructor
-     */
-    init: function () {
-        this._super.apply(this, arguments);
-
+export class SearchBar extends Interaction {
+    static selector = ".o_searchbar_form";
+    dynamicContent = {
+        _root: {
+            "t-on-focusout": this.debounced(this.onFocusOut, 100),
+            "t-on-safarihack": (ev) => {
+                this.linkHasFocus = ev.detail.linkHasFocus;
+            },
+        },
+        ".search-query": {
+            "t-on-input": this.debounced(this.onInput, 400),
+            "t-on-keydown": this.onKeydown,
+            "t-on-search": this.onSearch,
+        },
+    };
+    autocompleteMinWidth = 300;
+
+    setup() {
         this.keepLast = new KeepLast();
-
-        this._onInput = debounce(this._onInput, 400);
-        this._onFocusOut = debounce(this._onFocusOut, 100);
-    },
-    /**
-     * @override
-     */
-    start: function () {
-        this.$input = this.$('.search-query');
-
-        this.searchType = this.$input.data('searchType');
-        this.order = this.$('.o_search_order_by').val();
-        this.limit = parseInt(this.$input.data('limit'));
-        this.displayDescription = this.$input.data('displayDescription');
-        this.displayExtraLink = this.$input.data('displayExtraLink');
-        this.displayDetail = this.$input.data('displayDetail');
-        this.displayImage = this.$input.data('displayImage');
-        this.wasEmpty = !this.$input.val();
+        this.inputEl = this.el.querySelector(".search-query");
+        this.menuEl = null;
+        this.searchType = this.inputEl.dataset.searchType;
+        const orderByEl = this.el.querySelector(".o_search_order_by");
+        this.order = orderByEl.value;
+        this.limit = parseInt(this.inputEl.dataset.limit);
+        this.displayDescription = this.inputEl.dataset.displayDescription;
+        this.displayExtraLink = this.inputEl.dataset.displayExtraLink;
+        this.displayDetail = this.inputEl.dataset.displayDetail;
+        this.displayImage = this.inputEl.dataset.displayImage;
+        this.wasEmpty = !this.inputEl.value;
+        this.linkHasFocus = false;
         // Make it easy for customization to disable fuzzy matching on specific searchboxes
-        this.allowFuzzy = !this.$input.data('noFuzzy');
+        this.allowFuzzy = !this.inputEl.dataset.noFuzzy;
         if (this.limit) {
-            this.$input.attr('autocomplete', 'off');
+            this.inputEl.setAttribute("autocomplete", "off");
         }
 
         this.options = {
-            'displayImage': this.displayImage,
-            'displayDescription': this.displayDescription,
-            'displayExtraLink': this.displayExtraLink,
-            'displayDetail': this.displayDetail,
-            'allowFuzzy': this.allowFuzzy,
+            "displayImage": this.displayImage,
```

Comment:
i guess you can simply define these values here, it seems like you don't need `this.displayImage` and the others:
```js
const dataset = this.inputEl.dataset;
this.options = {
  "displayImage": dataset.displayImage,
  "displayDescription": dataset.displayDescription,
  "displayExtraLink": dataset.displayExtraLink,
  "displayDetail": dataset.displayDetail,
  "allowFuzzy": dataset.noFuzzy,
};
```